xbmc/addons/InputStream.cpp

```diff
@@ -96,7 +96,8 @@ bool CInputStream::Supports(CFileItem &fileitem)
     if (path.empty())
       continue;
 
-    if (fileitem.GetPath().compare(0, path.length(), path) == 0)
+    CRegExp r(true, CRegExp::asciiOnly, path.c_str());
+    if(r.RegFind(fileitem.GetPath())==0)
```

Comment:
not sure if this is right. this would even match a single character in the middle of the path.